pkg/skaffold/initializer/init.go

```diff
@@ -541,35 +542,73 @@ func printAnalyzeJSON(out io.Writer, skipBuild bool, pairs []builderImagePair, u
 func walk(dir string, force, enableJibInit bool, validateBuildFile func(bool, string) ([]InitBuilder, error)) ([]string, []InitBuilder, error) {
 	var potentialConfigs []string
 	var foundBuilders []InitBuilder
-	err := filepath.Walk(dir, func(path string, f os.FileInfo, e error) error {
-		if f.IsDir() && util.IsHiddenDir(f.Name()) {
-			logrus.Debugf("skip walking hidden dir %s", f.Name())
-			return filepath.SkipDir
-		}
-		if f.IsDir() || util.IsHiddenFile(f.Name()) {
-			return nil
+
+	var dirCallback func(path string, findBuilders bool) error
+	dirCallback = func(path string, findBuilders bool) error {
+		dirents, err := godirwalk.ReadDirents(path, nil)
+		if err != nil {
+			return err
 		}
-		if IsSkaffoldConfig(path) {
-			if !force {
-				return fmt.Errorf("pre-existing %s found", path)
+
+		var directories []*godirwalk.Dirent
+		findBuildersInDirectories := true
+		sort.Sort(dirents)
+
+		// Traverse files
+		for _, file := range dirents {
```

Comment:
this func is too long and hard to read. Can you reduce the number of branches or break it down further into smaller functions?.

Reply:
I'm trying now, but so far every attempt to break it apart just leads to passing lots of arguments between the functions, and it ends up being harder to follow. I'll keep trying, but so far inlining it seems the best in my opinion.